src/Menu.js

```diff
@@ -39,16 +54,29 @@ module.exports = class extends React.Component {
 
   componentWillUnmount() {
     if (this.tapListener) this.tapListener.remove();
+    this.removeTapStartListener();
     this.context.ambManager.destroy();
   }
 
+  removeTapStartListener() {
+    if (this.document) {
+      this.document.removeEventListener('mousedown', this.handleTapStart);
+      this.document.removeEventListener('touchstart', this.handleTapStart);
+    }
+  }
+
   addTapListener = () => {
-    const el = ReactDOM.findDOMNode(this);
-    if (!el) return;
-    const doc = el.ownerDocument;
-    if (!doc) return;
-    this.tapListener = createTapListener(doc.documentElement, this.handleTap);
-  };
+    if (this.document) {
+      this.tapListener = createTapListener(this.document, this.handleTap);
+    }
+  }
+
+  addTapStartListener = () => {
+    if (this.document) {
+      this.document.addEventListener('mousedown', this.handleTapStart);
+      this.document.addEventListener('touchstart', this.handleTapStart);
```

Comment:
Why use these event listener instead of the tap listener that was created at line 70?